apps/studio/src/features/editing-experience/components/form-builder/FormBuilder.tsx

```diff
@@ -53,32 +56,39 @@ const renderers: JsonFormsRendererRegistryEntry[] = [
 ]
 
 export interface FormBuilderProps {
-  component: keyof typeof IsomerSchema.components.complex
+  component: IsomerComplexComponentProps['type']
 }
 
 export default function FormBuilder({
   component,
 }: FormBuilderProps): JSX.Element {
-  const { properties, ...rest } = IsomerSchema.components.complex[component]
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const { type, ...props } = properties
-  const schema = {
-    ...rest,
-    properties: props,
-    components: IsomerSchema.components,
-  }
-
+  const [jsonSchema, setJsonSchema] = useState<IsomerJsonSchema | null>(null)
   const [formData, setFormData] = useState({})
 
+  trpc.page.getIsomerJsonSchema.useQuery(undefined, {
+    onSuccess({ schema }) {
+      const { properties, ...rest } = schema.components.complex[component]
+      const { type: _, ...props } = properties
```

Comment:
not an issue with this pr, but `type` is a keyword so we'll have alot of renaming down the line. easier to solve it by just renaming the property to `kind` or `variant` or smth

Reply:
Removed in 71a18021a67d22a779bab79620dc28904bb1c7b1.